content/posts/2018-01-10-kotlin-null.md

```diff
@@ -0,0 +1,249 @@
+To actually have it the Java code should be annotated with so-called Nullability Annotations.
+The platform itself does not have enough introspection knowledge about if a value can be `null` or not.
+
+OK, so now we’ve established that we need annotations in the Java code to make Kotlin happy.
+In other words, the RxJava source code should be changed to be like this.
+
+```java
+public static <T> Observable<T> just(@NonNull T item) {
+    ObjectHelper.requireNonNull(item, "The item is null");
+    return RxJavaPlugins.onAssembly(new ObservableJust<T>(item));
+}
+```
+
+The difference is in the `@NonNull` annotation. The IDE will highlight the original code as an error,
+the Kotlin compiler will stop compilation... The End.
+
+It is totally another story about mutiple _different_ Nullability Annotations artifacts
+from JetBrains, FindBugs, The Checker Framework, Android Support Library and others,
+but [Jesse has it covered](https://medium.com/square-corner-blog/rolling-out-nullable-42dd823fbd89).
+
+# The Unicorn
+
+## The Problem
+
+### Java
+
+At least from my personal perspective, it is extremely naive to expect that every single Java library
+available will follow the lead of Nullability Annotations usage in their code.
+Asking people to adopt these is actually asking them to change their development culture,
+habits and even entire system designs. And sometimes the benefit is too small for the effort.
+
+Yes, it will help with the documentation and IDE inspections, not only with the Kotlin usage,
+but if you have a billion of public methods in a huge framework without any particular knowledge
+if a return value or an argument can be `null`...
+The situation becomes even worse with non-library and non-open source projects with deadlines and priorities.
+
+### Habits
+
+I’ve spent something about 1.5 years writing Kotlin-only code every single workday.
+Sometimes Python, sometimes Bash, but it does not change the picture.
+
+The human mindset adapts very well. Typing Kotlin code and thinking with Kotlin concepts every day
+_will_ make you forget about simple things — like the one we are discussing here — Java does not have
+nullability metadata out of the box. You will expect that everything you touch actually has it
+(a classic [Midas mistake](https://en.wikipedia.org/wiki/Midas#Golden_Touch)),
+because your shiny Kotlin codebase has it and you have a great confidence in that.
+
+Prepare for unforeseen consequences.
+
+## The Solution
+
+### The Obvious
+
+This is a hard one since it is based on a non-automated semantics mechanism — the mind.
+It is always better to remember when you are using Java code and when you are using the Kotlin one.
+
+Ask more _strange_ questions during a code review — _Are we guarded against this Java code?_
+It may sound dumb and people are often too lazy to check but it is better to raise a question instead of doing nothing.
+
+### Contributions
```

Comment:
I dont fully understand the point you are trying to make in this section („Contributions“).  

Perhabs you would like to talk about open source contributions in general? Worth its own blog post? Imho, the „Contributions“ part could be shrink down to just 1-2 sentences which makes the whole blog post a little bit shorter / less reading time (people are lazy readers).